### PR TITLE
Prep libjpeg-turbo for runtimecore

### PR DESCRIPTION
**Pull request by colingpowell**
Issue: https://devtopia.esri.com/runtime/home-one/issues/470

Update `libjpeg-turbo` build settings to make it usable in Runtime Core.

- [x] Library builds for all platforms: 
  - [x86 ](http://runtime-test.esri.com:8080/view/DEV/job/dev-vtest-3rdparty_libraries-interface/147/downstreambuildview/)
  - [x64](http://runtime-test.esri.com:8080/view/DEV/job/dev-vtest-3rdparty_libraries-interface/148/downstreambuildview/)
  - [arm](http://runtime-test.esri.com:8080/view/DEV/job/dev-vtest-3rdparty_libraries-interface/150/downstreambuildview/)
  - [arm64](http://runtime-test.esri.com:8080/view/DEV/job/dev-vtest-3rdparty_libraries-interface/150/downstreambuildview/)
- [x] runtimecore + tests build with this library installed (a few baselines will need to be updated): 
  - [x86 ](http://runtime-test.esri.com:8080/view/vTest/job/vtest_interface/3009/)
  - [x64](http://runtime-test.esri.com:8080/view/vTest/job/vtest_interface/3008/downstreambuildview/)
  - [arm](http://runtime-test.esri.com:8080/view/vTest/job/vtest_interface/3010/downstreambuildview/)
  - [arm64](http://runtime-test.esri.com:8080/view/vTest/job/vtest_interface/3007/downstreambuildview/)

**Build Config**
Config settings are located in `jconfig.h` and `jconfigint.h`, which are based on the `libjpeg-turbo` config files located in the `skia` library.

`cmake` was run locally to auto-generate these config files and compare them those ported from `skia`. The `cmake`-generated config files included the pair of `#if (defined _WIN32)` blocks that were added to our own `jconfig.h` in 8206c5c. 

**SIMD Support**
SIMD support (the `turbo` in `libjpeg-turbo`) has been added to `arm` and `arm64`-equipped iOS and Android devices. `libjpeg-turbo` requires the `NASM` assembler for SIMD support on `x86` and `x64` architectures, and it seemed best to avoid introducing another binary into our 3rd party ecosystem solely to assemble native SIMD commands for this library. 

SIMD support is not included for WinUWP platforms, as VS projects don't presently allow files to be conditionally included in a project based on its platform.

For platforms that support SIMD, source files in `simd/[architecture]` are compiled, which contain SIMD-specific versions of various JPEG compression and decompression algorithms. Otherwise, `jsimd_none.c` is compiled, which defines these SIMD-specific functions as empty stubs.

All of this allows for the correct compression/decompression functions to be selected at runtime - for example:
```
#ifdef DCT_IFAST_SUPPORTED
      case JDCT_IFAST:
        if (jsimd_can_idct_ifast())
          method_ptr = jsimd_idct_ifast;
        else
          method_ptr = jpeg_idct_ifast;
        method = JDCT_IFAST;
        break;
#endif
```
The `jsimd_can_` functions are defined in both `jsimd_none.c` and the `simd/[architecture]` source files. They work essentially like flags - for example: 
```
GLOBAL(int)
jsimd_can_idct_ifast(void)
{
  return 0;
}
```
Where this particular functionality is allowed, this function is defined to return 1. 

Our build system appears to disallow building from identically-named source files located in different directories. Unfortunately, the source files located in each `/simd/[architecture]` directory follow this pattern. To work-around this limitation, we created a pair of uniquely-named source files in each directory and simply `#include`d the offending files there.  

Please review: @chri7325 @john4744 @dkalowsk

### CODE REVIEW

**Review by john4744 — APPROVED** [review]
looks good 👍

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+Makefile
```
Comment (chri7325) [inline .gitignore:5]: Can this be the RTC one so it'll help with all platforms? https://devtopia.esri.com/3rdparty/zlib/blob/runtimecore/.gitignore

File: libjpeg-turbo.lua [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+    "./",
+  }
+
+  -- -------------------------------------------------------------
+  -- configurations
+  -- -------------------------------------------------------------
+
+  if (os.is("windows")) then
+
+    defines {
```
Comment (chri7325) [inline libjpeg-turbo.lua]: Can the defins and files move to the appropriate windows blocks?

File: libjpeg-turbo.lua [hunk 3]
```diff
@@ -0,0 +1,606 @@
...
+    -- common configuration settings
+
+    dofile (_BUILD_DIR .. "/static_ios.lua")
+
+    -- project specific configuration settings
+
+    configuration { "ios*" }
+
+      defines {
+        "_REENTRANT",
```
Comment (chri7325) [inline libjpeg-turbo.lua]: Are these defines needed?

File: libjpeg-turbo.lua [hunk 4]
```diff
@@ -0,0 +1,606 @@
...
+    -- common configuration settings
+
+    dofile (_BUILD_DIR .. "/static_android.lua")
+
+    -- project specific configuration settings
+
+    configuration { "android*" }
+
+      defines {
+        "HAVE_MEMMOVE",
```
Comment (chri7325) [inline libjpeg-turbo.lua]: Is this define needed?

File: libjpeg-turbo.lua [hunk 5]
```diff
@@ -0,0 +1,606 @@
...
+  if (_TARGET_IS_WINUWP) then
+    -- -------------------------------------------------------------
+    -- configuration { "winuwp" } == _TARGET_IS_WINUWP
+    -- -------------------------------------------------------------
+
+    -- common configuration settings
+
+    dofile (_BUILD_DIR .. "/static_winuwp.lua")
+
+    defines {
```
Comment (chri7325) [inline libjpeg-turbo.lua]: Can this be tabbed over? It should also go after the configuration below

File: libjpeg-turbo.lua [hunk 6]
```diff
@@ -0,0 +1,606 @@
...
+  -- -------------------------------------------------------------
+
+  if (os.is("windows")) then
+
+    defines {
+      "TURBO_FOR_WINDOWS",
+    }
+
+    files {
+      "jsimd_none.c",
```
Comment (chri7325) [inline libjpeg-turbo.lua]: Can the jsimd files be put into a table above so they can easily be updated in the future.  Like https://devtopia.esri.com/3rdparty/skia/blob/runtimecore/skia.lua#L344

**Review by chri7325 — APPROVED** [review]
Looks great!  Just a few minor suggestions.

**Comment by chri7325** [comment]
Oh, same request for adding the cmake command that was used to generate the files in the commit that brought them in.

File: libjpeg-turbo.lua (hunk 3 above)
Comment (colingpowell) [reply libjpeg-turbo.lua]: Looks like these (and the android defines) were brought over from the old `jpeg.lua`...I'll remove them and confirm that everything still builds.

**Review by john4744 — APPROVED** [review]

**Review by chri7325 — APPROVED** [review]
Looks great!

**Review by dkalowsk — APPROVED** [review]

**Pull request merged**